### PR TITLE
[Merged by Bors] - feat: further theory about the Killing form of Lie algebras

### PR DESCRIPTION
**Pull request by ocfnash**
---

- [x] depends on: #6588

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Lie/Killing.lean [hunk 1]
```diff
@@ -70,15 +73,123 @@ lemma traceForm_apply_lie_apply (x y z : L) :
...
       ← LinearMap.mul_eq_comp]
 
+@[simp] lemma traceForm_eq_zero_of_isNilpotent [IsReduced R] [IsNilpotent R L M] :
+    traceForm R L M = 0 := by
+  ext x y
+  simp only [traceForm_apply_apply, LinearMap.zero_apply, ← isNilpotent_iff_eq_zero]
+  apply LinearMap.isNilpotent_trace_of_isNilpotent
+  exact isNilpotent_toEndomorphism_of_isNilpotent₂ R L M x y
+
+/-- This is barely worth having since it follows from `LieModule.traceForm_eq_zero_of_isNilpotent`
```
Comment (jcommelin) [inline Mathlib/Algebra/Lie/Killing.lean:83]: Do you want this comment in the docstring? Or just in a regular comment?

File: Mathlib/Algebra/Lie/Killing.lean [hunk 2]
```diff
@@ -70,15 +73,123 @@ lemma traceForm_apply_lie_apply (x y z : L) :
...
+class IsKilling : Prop :=
+  killingCompl_top_eq_bot : LieIdeal.killingCompl R L ⊤ = ⊥
+
+/-- The converse of this is true over a field of characteristic zero. There are counterexamples
+over fields with positive characteristic. -/
+instance IsKilling.isSemisimple [IsKilling R L] : IsSemisimple R L := by
+  refine' (isSemisimple_iff_no_abelian_ideals R L).mpr fun I hI ↦ _
+  rw [eq_bot_iff, ← IsKilling.killingCompl_top_eq_bot]
+  exact I.le_killingCompl_top_of_isLieAbelian
+
```
Comment (jcommelin) [inline Mathlib/Algebra/Lie/Killing.lean:192]:
```suggestion

-- TODO: formalize a positive-characteristic counterexample to the above instance

```

**Review by jcommelin — APPROVED** [review]
LGTM

bors d+

File: Mathlib/Algebra/Lie/Killing.lean (hunk 1 above)
Comment (ocfnash) [reply Mathlib/Algebra/Lie/Killing.lean:83]: Good point, I don't want this to be a docstring!

**Comment by ocfnash** [comment]
bors merge

**Pull request closed without merging**